File: backend_phyton/api/friends.py

from flask import Blueprint, request, jsonify
from adapters.firebase_adapter import verify_id_token, lacations, remove_friend
from adapters.myError import MyError

friends_blueprint = Blueprint('friends', __name__)

@friends_blueprint.route('/api/friends/<int:id>', methods=['DELETE'])
def delete_friend(id):
    token = request.headers.get('Authorization')
    try:
        decoded_token = verify_id_token(token,"Unauthorized request")
        user_id = decoded_token(token)['uid']

        remove_friend(user_id,id)

        return jsonify({"message": "Friend deleted successfully"}), 200
    except MyError as e:
        return jsonify({"error": e.message}),e.number

@friends_blueprint.route('/api/friends', methods=['GET'])
def get_friends_locations():
    token = request.headers.get('Authorization')
    try:
        decoded_token  = verify_id_token(token,"Unauthorized request")
        user_id = decoded_token(token)['uid']

        friend_locations = lacations(user_id)
        return jsonify(friend_locations),200
    except MyError as e:
        return jsonify({"error": e.message}),e.number
    except Exception as e:
        return jsonify("error":str(e) ),500

File: backend-dart/bin/routes/friends.dart
import 'dart:convert';

import 'package:shelf/shelf.dart';
import 'package:shelf_router/shelf_router.dart';

import '../adapter/firebaseAdapter.dart';
import 'responses.dart';

class Friends {

  static Future<Response> deleteFriend(Request request, String id) async {
    final userData = await FirebaseAdapter.verifyHeader(request.headers['authorization']);

    if(userData.$2 != 200) {
      return error(userData.$2,userData.$1);
    }

    await FirebaseAdapter.deleteFriend(userData.$1,id);

    return Response.ok('Friend deleted successfully');
  }

  static Future<Response> getFriendsLocations(Request request) async {
    final userData = await FirebaseAdapter.verifyHeader(request.headers['authorization']);

    if(userData.$2 != 200) {
      return error(userData.$2,userData.$1);
    }

    List<String> data = await FirebaseAdapter.friendLocations(userData.$1);

    return Response.ok(jsonEncode(data),headers: {'Content-Type': 'application/json'});
  }
  
}

final friendsRouter = Router()
  ..delete('/<id>',Friends.deleteFriend)
  ..get('/',Friends.getFriendsLocations);